src/DeploymentProvider/Rsync.php

```diff
@@ -99,44 +102,84 @@ public function configure(InputInterface $input, OutputInterface $output)
             putenv("RSYNC_RSH=sshpass -e ssh");
         }
     }
+
     /**
-     * @{inheritDoc}
+     * Beam up will beam to all target hosts
+     *
+     * @param Closure          $output
+     * @param bool             $dryrun
+     * @param DeploymentResult $deploymentResult
+     * @return DeploymentResult
+     * @throws RuntimeException
      */
-    public function up(\Closure $output = null, $dryrun = false, DeploymentResult $deploymentResult = null)
+    public function up(Closure $output = null, $dryrun = false, DeploymentResult $deploymentResult = null)
     {
-        return $this->deploy(
-            $this->buildCommand(
-                $this->beam->getLocalPath(),
-                $this->getTargetPath(),
-                $dryrun
-            ),
-            $output
-        );
+        /** @var DeploymentResult $mergedResult */
+        $mergedResult = null;
+        $results = [];
+        foreach ($this->getTargetPaths() as $server => $targetPath) {
+            $result = $this->deploy(
+                $this->buildCommand(
+                    $this->beam->getLocalPath(),
+                    $targetPath,
+                    $dryrun
+                ),
+                $output,
+                true // silence per-server output
+            );
```

Comment:
Can single server deployments please keep the original behaviour of streaming output directly? This re-introduces a delay that we specifically got rid of.

(It's fine to keep the delay for multi-server deployments for now though)

Reply:
Good idea actually; I can make it conditional.

Reply:
Sorry it's been a year https://github.com/heyday/beam/pull/63